.github/workflows/dev-deploy.yaml

```
@@ -47,41 +50,35 @@ jobs:
                   poetry install --no-interaction
 
             - name: Run pytest
-              if: steps.check_changes.outputs.changes_detected == 'true' || (github.event_name == 'workflow_dispatch' && github.event.inputs.force_deploy
-                  == 'true')
+              if: env.SHOULD_SETUP == 'true'
               run: poetry run pytest tests/
 
             - name: Cache pre-commit
-              if: (steps.check_changes.outputs.changes_detected == 'true' || (github.event_name == 'workflow_dispatch' && github.event.inputs.force_deploy
-                  == 'true')) && steps.cache-pre-commit.outputs.cache-hit != 'true'
+              if: env.SHOULD_SETUP == 'true' && steps.cache-pre-commit.outputs.cache-hit != 'true'
```

actionlint (1.7.4)

property "cache-pre-commit" is not defined in object type {check_changes: {conclusion: string; outcome: string; outputs: {string => string}}}

(expression)